### PR TITLE
#158028336 Display signup errors on the frontend.

### PR DESCRIPTION
**Pull request by OlatunjiYso**
#### What does this PR do?
This pull request captures all user errors during signup and renders it on the fronted

#### Description of Task to be completed?
Throws errors on the frontend when users attempt to:
 - register with an invalid email
 - register with non-alphanumeric password less than 6 figures
- username less than two letters

#### How should this be manually tested?
- clone the repo
- install all dependencies and run <npm start:dev>
- navigate to _http://localhost:3000/#!/signup_ 
- try signing up with some invalid parameters

#### What are the relevant pivotal tracker stories?
#158028336

#### Screenshots
screen shot 2018-06-08 at 4 20 55 pm
screen shot 2018-06-08 at 4 22 41 pm

### CODE REVIEW

File: config/middlewares/validations.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @description - Validate user signup
+ * @param {object} req - request body
+ * @param {object} res - response body
+ *
+ * @return {undefined} depends on user input
+ */
+exports.validateSignup = (req, res, next) => {
+  // Finds the validation errors in this request and wraps them in an object with handy functions
+  const errors = validationResult(req);
+  console.log(errors);
```
Comment (fortunee) [inline config/middlewares/validations.js]: Avoid leaving `logs` on your code. Please get rid of it @OlatunjiYso

File: config/middlewares/validations.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * @description - Validate user signup
+ * @param {object} req - request body
+ * @param {object} res - response body
+ *
+ * @return {undefined} depends on user input
+ */
+exports.validateSignup = (req, res, next) => {
+  // Finds the validation errors in this request and wraps them in an object with handy functions
+  const errors = validationResult(req);
```
Comment (fortunee) [inline config/middlewares/validations.js:11]: @OlatunjiYso Why are we validating the entire `req` object? 🤔  
I'm sure we're only expecting users input at `req.body` and not the entire `req` object.

File: public/js/controllers/index.js [hunk 3]
```diff
@@ -60,6 +61,11 @@ angular.module('mean.system').controller('IndexController', [
             localStorage.setItem('token', response.data.token);
          //  $http.default.headers.common['x-access-token'] = response.data.token;
```
Comment (fortunee) [inline public/js/controllers/index.js]: Not exactly your work but please remove this commented out code. @OlatunjiYso

**Review by fortunee — COMMENTED** [review]
Good job @OlatunjiYso Please review the comments and refactor. 
Also, try to adhere to our lint rules

File: config/middlewares/validations.js (hunk 2 above)
Comment (OlatunjiYso) [reply config/middlewares/validations.js:11]: Thank you @fortunee.
please, I seem not to have control over the validator API, 
I did it according to the [documentation](https://express-validator.github.io/docs/index.html)
what is being referenced eventually is `req.body.` ... what is set.

File: public/js/controllers/index.js (hunk 3 above)
Comment (OlatunjiYso) [reply public/js/controllers/index.js]: It has been removed.

File: config/middlewares/validations.js (hunk 1 above)
Comment (OlatunjiYso) [reply config/middlewares/validations.js]: Thanks, It has been removed.

File: config/middlewares/validations.js (hunk 2 above)
Comment (fortunee) [reply config/middlewares/validations.js:11]: Argh! I see! 😮 
Thanks for pointing me in the right direction @OlatunjiYso 👍

**Review by fortunee — APPROVED** [review]
We're good for now @OlatunjiYso Thanks

**Pull request merged**